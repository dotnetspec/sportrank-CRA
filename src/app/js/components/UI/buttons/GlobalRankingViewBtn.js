import React, { Component } from 'react';
import { Button } from 'react-bootstrap';
/**
 * Class representing global ranking view button in the BootstrapTable.
 * The component rendering in this area is controlled by
 * an @external "BrowserRouter"[? tbc]
 *
 * @extends React.Component
 */
class GlobalRankingViewBtn extends Component {

  //#region Constructor
  constructor(props, context) {
    super(props, context);
    // initial state
    this.state = {
      //GlobalRankingViewBtn_clicked: false
    };
  }

  // _handleChangeStatusPlayer(username) {
  //   if(username !== null){
  //     this.setState({GlobalRankingViewBtn_clicked: true });
  //     if(this.state.PlayerActive === true){
  //       this.setState({PlayerActive: false });
  //     }else{
  //       this.setState({PlayerActive: true });
  //     }
  //     //REVIEW: passing name to json to be deactivated
  //     //could be done in this component?
  //     this.props.history.push('/delete/@' + username);
  //   }
  //   else {
  //     console.log('no username passed to deactivate btn!')
  //   }
  // }

  onClickRankingViewSelected(cell, row, rowIndex){
    //REVIEW: the naming here.
    //tell Header that the view btn has been clicked
    //so it can display the 'Activate?' btn
    this.props.onChildClick();
    //console.log('Product #', rowIndex);
    // selectRowPropAfterClickRow.selectedRankingId = `${row['RANKINGID']}`;
    // console.log('selectRowPropAfterClickRow.selectedRankingId', selectRowPropAfterClickRow.selectedRankingId)
    // newrankIdCB(selectRowPropAfterClickRow.selectedRankingId);
    // viewingOnlyCB(true);
    this.props.onAfterUserUpdate();
    this.props.history.push('/home/@' + this.props.user.username);
    //this.openResultModal();
   }

  render () {
    //NB: this.props.user or .username etc. causing much confusion
    //needs to be sorted to be consistent
    return (
      // <Button className='globalrankingviewbtn'  data-cy='globalrankingviewbtn'
      // //bsStyle={this.state.bsStyle}
      // onClick={(e) => this._handleChangeStatusPlayer(this.props.user)}
      // >
      //   {this.state.btnText}
      // </Button>

      <Button
         className='globalrankingviewbtn'  data-cy='globalrankingviewbtn'
         //type="button"
         // onClick={() =>
         // this.onClickRankingViewSelected(this.props.cell, this.props.row, this.props.rowIndex)}
         //onClick={(e) => this._handleChangeStatusPlayer(this.props.user)}
      >
      {this.props.cell} {this.props.row} {this.props.rowIndex}
      View
      </Button>
    )
  }
  //#endregion
}
export default GlobalRankingViewBtn
