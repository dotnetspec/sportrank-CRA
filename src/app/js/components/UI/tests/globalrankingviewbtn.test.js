import {
  //wait,
} from '@testing-library/dom'
// adds special assertions like toHaveTextContent
import 'jest-dom/extend-expect'
import GlobalRankingViewBtn from '../buttons/GlobalRankingViewBtn'
import React from 'react'
import { render, fireEvent, cleanup } from '@testing-library/react'

//originally based on example:
//https://testing-library.com/docs/dom-testing-library/example-intro

//this is the 'View' btn on the initial glabol rnakings page

afterEach(cleanup);

it('calls "onClick" prop on button click', () => {
  // Render new instance in every test to prevent leaking state
  const historyMock = { push: jest.fn() };
  const onClick = jest.fn();
  const onAfterUserUpdate = jest.fn();
  const newrankId = jest.fn();
  const setviewingOnlyCB = jest.fn();
  const setnewrankIdCB = jest.fn();

  const row = {RANKINGNAME: "mplayer1rank", RANKINGDESC: "mp1r", ACTIVE: true, RANKINGID: "5c875c79adeb832d3ec6732d"}
  const { getByText } = render(<GlobalRankingViewBtn
    setspecificRankingOptionBtnsCB={onClick}
    row={row}
    onAfterUserUpdate={onAfterUserUpdate}
    newrankId={newrankId}
    setviewingOnlyCB={setviewingOnlyCB}
    history={historyMock}
    setnewrankIdCB={setnewrankIdCB}
    />);

  fireEvent.click(getByText(/View/i));
  expect(onClick).toHaveBeenCalled();
  //expect(onAfterUserUpdate).toHaveBeenCalled();
  //expect(newrankId).toHaveBeenCalled();
  expect(setviewingOnlyCB).toHaveBeenCalled();
  expect(setnewrankIdCB).toHaveBeenCalled();
});
