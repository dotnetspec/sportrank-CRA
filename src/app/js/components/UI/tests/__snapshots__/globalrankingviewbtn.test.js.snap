// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Global Ranking View Btn UI clicking table view btn 1`] = `ShallowWrapper {}`;
